sdk/textanalytics/Azure.AI.TextAnalytics/src/TextAnalyticsClient.cs

```diff
@@ -19,12 +19,28 @@ public partial class TextAnalyticsClient
         private readonly Uri _baseUri;
         private readonly HttpPipeline _pipeline;
         private readonly ClientDiagnostics _clientDiagnostics;
-        private readonly string _subscriptionKey;
         private readonly string _apiVersion;
         private readonly TextAnalyticsClientOptions _options;
 
         private readonly string DefaultCognitiveScope = "https://cognitiveservices.azure.com/.default";
 
+        private string _subscriptionKey;
+
+        private string SubscriptionKey
```

Comment:
Could we make this a separate class as well, the way storage does with SharedKeyCredential?

Reply:
I'm wondering if we should follow storage's lead here by making the credential a separate class, or if we should try to keep authentication by subscription key simple with just having the user provide a string. Am I missing a reason for why it should be a separate class?

Reply:
Because other cognitive services use the same auth methods, better to make it a separate class and then see how we make it in a shared way (shared package, etc) that other services will consume it too.